casper/src/test/scala/coop/rchain/casper/util/ProtoUtilTest.scala

```diff
@@ -24,4 +27,59 @@ class ProtoUtilTest extends FlatSpec with Matchers with GeneratorDrivenPropertyC
       result should contain theSameElementsAs ((justificationsHashes ++ parentsHashes).toSet)
     }
   }
+
+  import MultiParentCasperTestUtil._
+
+  implicit val timeEff = new LogicalTime[Effect]
+
+  private val (validatorKeys, validatorPks) = (1 to 4).map(_ => Ed25519.newKeyPair).unzip
+  private val genesis = buildGenesis(
+    buildGenesisParameters(4, createBonds(validatorPks))
+  )
+
+  "unseenBlockHashes" should "return empty for a single block dag" in effectTest {
+    val node = HashSetCasperTestNode.standaloneEff(genesis, validatorKeys.head)
+    import node._
+    implicit val timeEff = new LogicalTime[Effect]
+
+    for {
+      deploy               <- ConstructDeploy.basicDeployData[Effect](0)
+      casper               = node.casperEff
+      _                    <- casper.deploy(deploy)
+      createBlockResult    <- casper.createBlock
+      Created(signedBlock) = createBlockResult
+      _                    <- casper.addBlock(signedBlock, ignoreDoppelgangerCheck[Effect])
+      dag                  <- casper.blockDag
+      unseenBlockHashes    <- ProtoUtil.unseenBlockHashes[Effect](dag, signedBlock)
+      _                    = unseenBlockHashes should be(Set.empty[BlockHash])
+      _                    <- node.tearDown()
+    } yield ()
+  }
+
+  it should "return all but the first block when passed the first block in a chain" in effectTest {
+    val node = HashSetCasperTestNode.standaloneEff(genesis, validatorKeys.head)
+    import node._
+    implicit val timeEff = new LogicalTime[Effect]
+
+    for {
+      deploy            <- ConstructDeploy.basicDeployData[Effect](0)
+      casper            = node.casperEff
+      _                 <- casper.deploy(deploy)
+      createBlockResult <- casper.createBlock
+      Created(block0)   = createBlockResult
+      _                 <- casper.addBlock(block0, ignoreDoppelgangerCheck[Effect])
+      _ <- (1 to 8).toList.traverse_[Effect, Unit] { i =>
+            for {
+              deploy            <- ConstructDeploy.basicDeployData[Effect](i)
+              createBlockResult <- casper.deploy(deploy) *> casper.createBlock
```

Comment:
I don't have a preference here but I've seen some discussion saying that we prefer `>>` instead of `*>`. Not a big deal though.